mismi-s3/src/Mismi/S3/Commands.hs

```diff
@@ -132,6 +134,37 @@ getSize :: Address -> AWS (Maybe Int)
 getSize a =
   headObject a >>= pure . maybe Nothing (^. A.horsContentLength)
 
+getSizeRecursively :: Address -> AWS [Sized Address]
+getSizeRecursively prefix =
+  getSizeRecursively' prefix $$ DC.consume
+
+getSizeRecursively' :: Address -> Source AWS (Sized Address)
+getSizeRecursively' prefix@(Address (Bucket b) (Key k)) =
```

Comment:
Tests for these woulld be good, see [examples](https://github.com/ambiata/mismi/blob/master/mismi-s3/test/Test/IO/Mismi/S3/Commands.hs#L104)